File: miniapp/src/api/post.ts
/**
 * 帖子 - 与 backend /api/v1/posts 一致
 */
import request, { type RequestOptions } from '@/utils/request'

const PREFIX = '/api/v1/posts'

export type PostType = 'complaint' | 'sharing' | 'question'

export interface PostItem {
  id: string
  user_id: string
  anonymous_name: string
  content: string
  images: string[] | null
  tags: string[] | null
  type: string
  salary_range: string | null
  industry: string | null
  city: string | null
  view_count: number
  like_count: number
  comment_count: number
  status: string
  risk_status: string
  risk_score: number | null
  risk_reason: string | null
  created_at: string
  updated_at: string
}

export interface PostCreateParams {
  content: string
  images?: string[] | null
  tags?: string[] | null
  type?: PostType
  salary_range?: string | null
  industry?: string | null
  city?: string | null
}

function queryString(params: Record<string, string | number>): string {
  const s = new URLSearchParams()
  Object.entries(params).forEach(([k, v]) => s.set(k, String(v)))
  const q = s.toString()
  return q ? `?${q}` : ''
}

/** 帖子列表（广场：热门/最新） */
export function getPostList(params: { sort: 'hot' | 'latest'; limit?: number; offset?: number }) {
  const { sort, limit = 20, offset = 0 } = params
  return request<PostItem[]>({
    url: `${PREFIX}${queryString({ sort, limit, offset })}`,
    method: 'GET',
  })
}

/** 帖子详情 */
export function getPostDetail(postId: string) {
  return request<PostItem>({ url: `${PREFIX}/${postId}`, method: 'GET' })
}

/** 发帖（发布后 risk_status=pending，异步风控） */
export function createPost(data: PostCreateParams, options?: Partial<RequestOptions>) {
  return request<PostItem>({ url: PREFIX, method: 'POST', data, ...options })
}

// ==================== 关注流 ====================

export interface FeedListRes {
  items: PostItem[]
  total: number
}

/** 获取关注流（我关注的人发布的帖子） */
export function getFeed(params?: { limit?: number; offset?: number }): Promise<FeedListRes> {
  const { limit = 20, offset = 0 } = params || {}
  return request<FeedListRes>({
    url: `/api/v1/user/me/feed?limit=${limit}&offset=${offset}`,
    method: 'GET',
  })
}

// ==================== 搜索 ====================

export interface SearchParams {
  keyword?: string
  tags?: string[]
  user_id?: string
  industry?: string
  city?: string
  salary_range?: string
  sort?: 'hot' | 'latest'
  limit?: number
  offset?: number
}

/** 搜索帖子 */
export function searchPosts(params: SearchParams): Promise<FeedListRes> {
  const {
    keyword,
    tags,
    user_id,
    industry,
    city,
    salary_range,
    sort = 'latest',
    limit = 20,
    offset = 0,
  } = params

  const query: Record<string, string | number> = { sort, limit, offset }
  if (keyword) query.keyword = keyword
  if (tags && tags.length > 0) query.tags = tags.join(',')
  if (user_id) query.user_id = user_id
  if (industry) query.industry = industry
  if (city) query.city = city
  if (salary_range) query.salary_range = salary_range

  return request<FeedListRes>({
    url: `${PREFIX}/search${queryString(query)}`,
    method: 'GET',
  })
}

File: backend/app/schemas/post.py
"""
帖子 - 请求/响应模型；与迭代规划 2.2、技术方案 2.2.1 一致
"""
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel, Field


class PostCreate(BaseModel):
    """发帖：anonymous_name 由服务端用当前用户匿名昵称填充"""
    content: str = Field(..., min_length=1, max_length=5000, description="帖子内容，1-5000字符")
    images: Optional[List[str]] = Field(None, max_length=9, description="图片列表，最多9张")
    tags: Optional[List[str]] = Field(None, max_length=10, description="标签列表，最多10个")
    type: Literal["complaint", "sharing", "question"] = "complaint"
    salary_range: Optional[str] = Field(None, max_length=20, description="工资区间")
    industry: Optional[str] = Field(None, max_length=50, description="行业")
    city: Optional[str] = Field(None, max_length=50, description="城市")
    visibility: Literal["public", "followers", "private"] = "public"


class PostResponse(BaseModel):
    id: str
    user_id: str
    anonymous_name: str
    content: str
    images: Optional[List[str]] = None
    tags: Optional[List[str]] = None
    type: str
    salary_range: Optional[str] = None
    industry: Optional[str] = None
    city: Optional[str] = None
    visibility: str
    view_count: int
    like_count: int
    comment_count: int
    status: str
    risk_status: str
    risk_score: Optional[int] = None
    risk_reason: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
